support-frontend/assets/pages/[countryGroupId]/checkout/components/PersonalDetailsFields.tsx

```diff
@@ -111,74 +113,78 @@ export function PersonalDetailsFields({
 				</div>
 			)}
 			{children}
-			<div>
-				<TextInput
-					id="firstName"
-					data-qm-masking="blocklist"
-					label="First name"
-					value={firstName}
-					autoComplete="given-name"
-					autoCapitalize="words"
-					onChange={(event) => {
-						setFirstName(event.target.value);
-					}}
-					onBlur={(event) => {
-						event.target.checkValidity();
-					}}
-					name="firstName"
-					required
-					maxLength={40}
-					error={firstNameError}
-					pattern={doesNotContainExtendedEmojiOrLeadingSpace}
-					onInvalid={(event) => {
-						preventDefaultValidityMessage(event.currentTarget);
-						const validityState = event.currentTarget.validity;
-						if (validityState.valid) {
-							setFirstNameError(undefined);
-						} else {
-							if (validityState.valueMissing) {
-								setFirstNameError('Please enter your first name.');
+			{firstName !== undefined && (
```

Comment:
Should we also check setFirstName has been passed in here too? eg.

```{firstName && setFirstName && <div> ... </div>}```

The reason I suggest this is do we really see a use case where we'd render the `firstName` field but not need a `setFirstName` handler in place too?

Reply:
Lets discuss whether reusing the `PersonalDetailsFields.tsx` component is the best approach here, it introduces complexity to the component, and a few new potential failure points, it might be simpler to stick with the old approach of using an inline `<TextInput>` for email fields in the one-time-checkout?

Reply:
Maybe `PersonalDetailsFields.tsx` component could contain a `PersonalEmailFields.tsx` component that could also be called directly from the `oneTimeCheckoutComponent` instead? Would that be a better solution? Then no need for additional `setFirstName` checks you mention above too?

Reply:
Have done the refactor listed above. It avoids using uneccesary `firstName`/`setFirstName` checks and is componentised better/simpler now too. Hope thats what you were suggesting.

Reply:
@paul-daniel-dempsey https://github.com/guardian/support-frontend/pull/6795#discussion_r1971410896, looks neat 👍